docs/assessment.md

```diff
@@ -128,6 +128,82 @@ By default, PrairieLearn provides an incentive for students to answer a specific
+- **Best current score** is the highest submission percentage score that the student has received since last achieving 100% for a submission. This is initialized to zero.
+- **Question value** is the current maximum number of points that the question is worth. This starts at `autoPoints` and increases by `autoPoints` for each consecutive correct submission. This is initialized to `autoPoints`.
+- **Question points** is the total number of points earned by the student for the question. This will never decrease. This is initialized to zero.
+
+For each submission attempt made by a student, the scoring algorithm is as follows:
+1. If the `submission percentage score` is more than `best current score`, add `(submission percentage score - best current score) / 100 * question value` to `question points`, capping `question points` at `maxAutoPoints`.
+2. If the `submission percentage score` is 100%, then reset `best current score` to zero. Otherwise, if the `submission percentage score` is more than `best current score`, then set `best current score` to `submission percentage score`.
+3. If the `submission percentage score` is 100%, then increase `question value` by `autoPoints`. Otherwise set `question value` to `autoPoints`. If `constantQuestionValue` is true then skip this step.
+
+As an example, suppose a student makes 8 submissions of varying degrees of correctness to a question. Assume the question has `autoPoints` set to 4 and `maxAutoPoints` set to 10. The student's submissions are as follows:
+
+Submission number | Submission perc. score | Best current score | Question value | Question points | Comment
+--- | --- | --- | --- | --- | ---
+Init | - | 0 | 4 | 0 | Initial values
+1 | 50% | 50% | 4 | 2 | (50% - 0%) * 4 = 2 points awarded
+2 | 80% | 80% | 4 | 3.2 | (80% - 50%) * 4 = 1.2 points awarded
+3 | 20% | 80% | 4 | 3.2 | No improvement in score, so no points awarded
+4 | 100% | 0% | 8 | 4 | (100% - 80%) * 4 = 0.8 points awarded, `best current score` reset to 0%, `question value` increased to 8
+5 | 50% | 50% | 4 | 8 | (50% - 0%) * 8 = 4 points awarded, `question value` reset to 4
+6 | 0% | 50% | 4 | 8 | No improvement in score, so no points awarded
+7 | 90% | 90% | 4 | 9.6 | (90% - 50%) * 4 = 1.6 points awarded
+8 | 100% | 0% | 8 | 10 | (100% - 90%) * 4 = 0.4 points awarded, `best current score` reset to 0%, `question value` increased to 8
+
+After the eight submissions above the student has achieved maximum points for the question. They can continue to answer the question for additional practice but they cannot earn any more points for it.
+
+## Question scoring for `Exam` assessments
+
+Assessments with `"type": "Exam"` are designed to test students' knowledge of course material. Students can retry questions for reduced points, so long as `allowRealTimeGrading` is true (the default). This section describes in detail how repeated attempts are scored, assuming that the question has `autoPoints` set to an array of values.
```

Comment:
I'm wondering if using "list" or "sequence" instead of "array" may be better for non-tech instructors? Both here and in other references?

Reply:
Done in 4191a4335164ac391314b4b9f76d725eba219f17. I changed all uses of `array` to `list` in this file. I didn't modify other doc files at the moment.